### DISCUSSION TITLE
How to load Ocelot routes from PostgreSQL database?

### DISCUSSION THREAD

**Discussion by aftabkajal** [ThreeMammals/Ocelot#2232] [Q&A] [answered]
Hello,

I am currently working on an Ocelot API Gateway and would like to implement a feature where the routes are dynamically loaded from a PostgreSQL database instead of the default ocelot.json configuration file.

Could you please guide me on how to approach this? Specifically, I need help with:

1. Setting up the database structure for storing route configurations.
2. Modifying Ocelot to fetch and use routes dynamically from the database.
3. Best practices for caching and optimizing performance in this setup.

If you have any resources, examples, or steps to follow, they would be greatly appreciated.

Thank you!

**Comment by raman-m** [comment, marked as answer]
Hi @aftabkajal !
You have the option to read from a database or manually create a `FileConfiguration` object prior to injecting it into the `AddOcelot` method.
### The answer is https://github.com/ThreeMammals/Ocelot/discussions/2217#discussioncomment-11372021
